src/moonlink/src/storage/mooncake_table/snapshot_read_output.rs

```diff
@@ -131,7 +207,8 @@ impl ReadOutput {
 
     /// Handle cleanup and notifications when resolving remote filepaths fails.
     async fn handle_resolution_error(&mut self, mut cache_handles: Vec<NonEvictableHandle>) {
-        let mut evicted_files_to_delete_on_error: Vec<String> = vec![];
+        // In most cases, we only have one remote file to delete, so pre-allocate a small vector.
+        let mut evicted_files_to_delete_on_error: Vec<String> = Vec::with_capacity(1);
```

Comment:
We know the overall vector size here (cache_handles + puffin_cache_handles + associated_files), so we could precisely pre-allocate?